src/plot_api/plot_api.js

```diff
@@ -552,7 +552,7 @@ function redraw(gd) {
     helpers.cleanLayout(gd.layout);
 
     gd.calcdata = undefined;
-    return exports.plot(gd).then(function() {
+    return exports._doPlot(gd).then(function() {
```

Comment:
Can't we drop `exports.` from these calls in this file?
```suggestion
    return _doPlot(gd).then(function() {
```

Reply:
I did try that.
But some tests failed.

Reply:
oh maybe the spies don't work then. OK, we can leave it.